### PR TITLE
Added setAddtionalParams method to externalROMLoader

### PR DESCRIPTION
**Pull request by AnthoneyGriffith**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
 #1856 

##### What are the significant changes in functionality due to this change request?
Added method to externalROMloader that has xml nodes as inputs for changing the pickledROM attributes of the ROM. Alongside this is a test showing the ability to change the clusteredEvalMode of the ROM with this new method.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [ ] 1. Review all computer code.
- [ ] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [ ] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [ ] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [ ] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [ ] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [ ] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [ ] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [ ] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

File: scripts/externalROMloader.py [hunk 1]
```diff
@@ -120,6 +120,21 @@ def __setstate__(self, d):
     self._frameworkLoc = d['whereFrameworkIs']
     self._load(self._binaryLoc, self._frameworkLoc)
 
+  def setAdditionalParams(self, nodes):
+    '''
```
Comment (dylanjm) [inline scripts/externalROMloader.py:124]: I think the convention is to use `"""` instead of `'''`.

File: scripts/externalROMloader.py [hunk 2]
```diff
@@ -120,6 +120,21 @@ def __setstate__(self, d):
...
+    Set ROM parameters for external evaluation
+    @ In, nodes, list, list of xml nodes for setting parameters of pickleROM
+    @ Out, None
+    '''
+    from ravenframework.SupervisedLearning.pickledROM import pickledROM
+    spec = pickledROM.getInputSpecification()()
```
Comment (dylanjm) [inline scripts/externalROMloader.py:130]: I'm not sure, but couldn't you just use `self.rom.getInputSpecification()` to get the input specs, or do you need the pickledROM input specs?

Comment (AnthoneyGriffith) [reply scripts/externalROMloader.py:130]: I believe it raises an error `AttributeError: 'ravenROMexternal' object has no attribute 'getInputSpecification'`

File: tests/framework/ROM/pickleTests/eRl_setAddtlParams.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+import externalROMloader
+
+def check(runner, before, after, signal):
+    '''
+    Decides of application of setAdditionalParams was successful
+    @ In, runner, externalROMloader object
+    @ In, before/after, list, list of xml objects 
+    @ In, signal, string, name of signal that exists in ROM
+    @ Out, string, 'Pass' or 'Fail'
+    '''
```
Comment (wangcj05) [inline tests/framework/ROM/pickleTests/eRl_setAddtlParams.py:41]: @AnthoneyGriffith could you follow the raven code standards? This applies to all the methods. This is required for any PR. You can check the link: https://github.com/idaholab/raven/wiki/RAVEN-Software-Commentary-Standard

**Review by wangcj05 — CHANGES_REQUESTED** [review]

File: tests/framework/ROM/pickleTests/eRl_setAddtlParams.py (hunk 3 above)
Comment (AnthoneyGriffith) [reply tests/framework/ROM/pickleTests/eRl_setAddtlParams.py]: Thank you for pointing those out! I believe I have changed all of them to the correct styling.

**Comment by PaulTalbot-INL** [comment]
We're set to update the submodule in RAVEN soon; do you think we'll have this merged soon so we can include it?

**Comment by PaulTalbot-INL** [comment]
We're set to update the submodule in RAVEN soon; do you think we'll have this merged soon so we can include it?

**Comment by wangcj05** [comment]
@PaulTalbot-INL @dylanjm anyone want to merge it?

**Comment by joshua-cogliati-inl** [comment]
Hm, how is tests/framework/ROM/pickleTests/externalROMloader/arma.pk generated?  At some point we will need to update it.

**Comment by dylanjm** [comment]
@joshua-cogliati-inl That's a good catch. @AnthoneyGriffith is there a way we can add the training XML and data to RAVEN?

**Pull request merged**